operations/service_deploy_smoke_utils.go

```diff
@@ -93,14 +95,57 @@ func getLatestGithubCommit() (string, error) {
 	return "", errors.New("could not find latest commit in response")
 }
 
-func checkTaskByCommit(username, key string) error {
+func checkContainerTask(username, key string) error {
+	client := utility.GetHTTPClient()
+	defer utility.PutHTTPClient(client)
+
+	var task apimodels.APITask
+OUTER:
+	for i := 0; i <= 30; i++ {
+		// check task
+		if i == 30 {
+			return errors.Errorf("task status is %s (expected %s)", task.Status, evergreen.TaskSucceeded)
+		}
+		time.Sleep(10 * time.Second)
+		grip.Infof("checking for task '%s' (%d/30)", containerTaskId, i+1)
+
+		var err error
+		task, err = checkTask(client, username, key, containerTaskId)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		if task.Status == evergreen.TaskFailed {
+			return errors.Errorf("task status is %s (expected %s)", task.Status, evergreen.TaskSucceeded)
+		}
+		if task.Status != evergreen.TaskSucceeded {
+			grip.Infof("found task is status %s", task.Status)
+			task = apimodels.APITask{
+				Status: task.Status,
+			}
+			continue OUTER
+		}
+
+		// retry for *slightly* delayed logger closing
+		err = checkLog(task, client, agent.PodMode, username, key)
+		if err != nil {
+			return err
+		}
+
+		grip.Info("Successfully checked container task")
+		return nil
```

Comment:
Seems like some of the copy-pasted for loop logic around task checking could be consolidated between `checkContainerTask` and `checkHostTaskByCommit` by creating a helper function that accepts the task ID + agent mode and returns the current task status.

Reply:
Done, refactored the task checking loop to a helper function.